tests/D2L.CodeStyle.Analyzers.Test/Specs/TypeDeclarationImmutabilityAnalyzer.cs

```diff
@@ -432,12 +443,12 @@ class SomeEventArgs { }
 
 
 
-		static /* NonImmutableTypeHeldByImmutable(Class, Object, ) */ Types.SomeImmutableGenericInterfaceGivenT</* NonImmutableTypeHeldByImmutable(Class, Object, ) */ object /**/, int> /**/ /* MemberIsNotReadOnly(Field, m_field, AnalyzedClassMarkedImmutable) */ m_field /**/;
-		static readonly /* NonImmutableTypeHeldByImmutable(Class, Object, ) */ Types.SomeImmutableGenericInterfaceGivenT</* NonImmutableTypeHeldByImmutable(Class, Object, ) */ object /**/, int> /**/ m_field;
-		/* NonImmutableTypeHeldByImmutable(Class, Object, ) */ Types.SomeImmutableGenericInterfaceGivenT</* NonImmutableTypeHeldByImmutable(Class, Object, ) */ object /**/, int> /**/ /* MemberIsNotReadOnly(Field, m_field, AnalyzedClassMarkedImmutable) */ m_field /**/;
-		readonly /* NonImmutableTypeHeldByImmutable(Class, Object, ) */ Types.SomeImmutableGenericInterfaceGivenT</* NonImmutableTypeHeldByImmutable(Class, Object, ) */ object /**/, int> /**/ m_field;
-		/* NonImmutableTypeHeldByImmutable(Class, Object, ) */ Types.SomeImmutableGenericInterfaceGivenT</* NonImmutableTypeHeldByImmutable(Class, Object, ) */ object /**/, int> /**/ Property { get; }
-		Types.SomeImmutableGenericInterfaceGivenT</* NonImmutableTypeHeldByImmutable(Class, Object, ) */ object /**/, int> Property { get { return default; } }
+		static /* NonImmutableTypeHeldByImmutable(Class, Object, ) */ Types.SomeImmutableGenericInterfaceGivenT<object, int> /**/ /* MemberIsNotReadOnly(Field, m_field, AnalyzedClassMarkedImmutable) */ m_field /**/;

```

Comment:
`[OnlyIf]` doesn't complain if a non-immutable type is passed to the parameter itself